### PR TITLE
fix(clerk-js): Use the correct returnBack url when GoogleOneTap remains open across navigations

### PR DESCRIPTION
**Pull request by panteliselef**
## Description

Bug fix: Use the correct returnBack url when GoogleOneTap remains open across navigations.
Previously it will only use the url that existed in the browser url bar at the time the component was initially rendered.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/clerk-js/src/ui/components/GoogleOneTap/one-tap-start.tsx [hunk 1]
```diff
@@ -16,35 +16,14 @@ function _OneTapStart(): JSX.Element | null {
...
-          signUpUrl,
-          continueSignUpUrl,
-          secondFactorUrl,
-          firstFactorUrl,
-          signUpForceRedirectUrl,
-          signInForceRedirectUrl,
-        },
-        navigate,
-      );
+      await clerk.handleGoogleOneTapCallback(res, ctx.generateCallbackUrls(window.location.href), navigate);
```
Comment (panteliselef) [inline packages/clerk-js/src/ui/components/GoogleOneTap/one-tap-start.tsx:26]: ❗️I couldn't find a robust way to listen to url changes (history) so i went with the synchronous solution to pass the current url when the callback runs

Comment (desiprisg) [reply packages/clerk-js/src/ui/components/GoogleOneTap/one-tap-start.tsx:26]: Can we do something like:
```
window.history.pushState = new Proxy(window.history.pushState, {
  apply: (target, thisArg, argArray) => {
    // trigger here what you need
    return target.apply(thisArg, argArray);
  },
});
```

**Review by desiprisg — APPROVED** [review]
Looks good but let's try to do it with a proxy on `pushState` and see if that simplifies things?!

**Comment by panteliselef** [comment]
We will attempt to find another solution in another PR

**Pull request merged**